server/src/main/java/org/elasticsearch/transport/TransportMessage.java

```diff
@@ -39,6 +39,12 @@ public TransportAddress remoteAddress() {
         return remoteAddress;
     }
 
+    public TransportMessage() {
```

Comment:
Hmmm. If it doesn't need `StreamInput` to be read, maybe don't define either ctor and just use the empty ctor that comes for free. But I see the point of defining this to sort of tell people that it is useful and that subclasses are going to have it. Either way I think some javadoc would be nice.